sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/tests/Tests/MigrationSapDiscoveryTests.cs

```diff
@@ -0,0 +1,333 @@
+        private Azure.Storage.Blobs.BlobClient GetBlobContentClient(string sasUri)
+        {
+            // Split the sas uri into blob uri and sas token.
+            // https://learn.microsoft.com/en-us/azure/ai-services/translator/document-translation/how-to-guides/create-sas-tokens?tabs=blobs#use-your-sas-url-to-grant-access-to-blob-storage
+            var sasUriFragment = sasUri.Split('?');
+            // Checks if the sas uri fragments are not-empty, not-null and has 2 elements.
+            if (sasUriFragment.Length != 2)
+            {
+                throw new InvalidDataException(
+                    $"Invalid sas uri: {sasUri}. Sas uri should be in the format of <blobUri>?<sasToken>");
+            }
+
+            // Construct the blob client with a sas token.
+            var blobClient = new Azure.Storage.Blobs.BlobClient(
+                new Uri(sasUriFragment[0]),
+                new AzureSasCredential(sasUriFragment[1]));
+
+            return blobClient;
+        }
+
+        /// <summary>
+        /// Tracks till it satisfies the input condition (boolean function).
+        /// </summary>
+        /// <param name="func">The function describing the condition was achieved or not.</param>
+        /// <param name="totalCount">The total count to loop.</param>
+        /// <returns>The status on whether the condition was achieved.</returns>
+        public static async Task<bool> TrackTillConditionReachedForAsyncOperationAsync(
+            Func<Task<bool>> func,
+            int totalCount = 100)
+        {
+            var isConditionReached = false;
+            int counter = 0;
+            while (counter < totalCount)
+            {
+                await Task.Delay(TimeSpan.FromSeconds(2));
+                if (await func())
+                {
+                    isConditionReached = true;
+                    break;
+                }
+
+                counter++;
+            }
+
+            return isConditionReached;
+        }
+
+        /// <summary>
+        /// Track for input sas uri for excel upload.
+        /// </summary>
+        /// <param name="operation">Operation to poll.</param>
+        /// <returns>If uri is now available to fetch.</returns>
+        protected async Task<bool> TrackForDiscoveryExcelInputSasUriAsync(ArmOperation<OperationStatusResult> operation)
+        {
+            var operationStatus = await Client.GetGenericResources().GetAsync(ResourceIdentifier.Parse(operation.Value.Id));
+            var map = JObject.Parse(operationStatus?.GetRawResponse()?.Content?.ToString());
+            var opProperties = map?["properties"];
+            Assert.IsNotNull(opProperties);
+
+            var status = opProperties?["status"];
+            Assert.IsNotNull(status);
+
+            if (status.ToString() == ImportOperationState.AwaitingFile.ToString()
+                && !string.IsNullOrEmpty(opProperties?["discoveryExcelSasUri"].ToString()))
+            {
+                return true;
```

Comment:
let's check with arthur to get a permanent model for this where we can pasrse model